homeassistant/components/samsungtv/__init__.py

```diff
@@ -1 +1,53 @@
 """The Samsung TV integration."""
+import socket
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_ON_ACTION, DEFAULT_NAME, DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            cv.ensure_list,
+            [
+                vol.Schema(
+                    {
+                        vol.Required(CONF_HOST): cv.string,
+                        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                        vol.Optional(CONF_PORT): cv.port,
+                        vol.Optional(CONF_ON_ACTION): cv.SCRIPT_SCHEMA,
+                    }
+                )
+            ],
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Samsung TV integration."""
+    if DOMAIN in config:
+        hosts = []
+        for entry_config in config[DOMAIN]:
+            host = socket.gethostbyname(entry_config[CONF_HOST])
+            if host not in hosts:
```

Comment:
This should be checked already in config validation.

Reply:
Ahh, now I understand what you mean.